### ISSUE TITLE
get_diagnostics omits IsSuppressed (pragma/suppressor-suppressed diagnostics invisible or unmarked)

### ISSUE THREAD

**Issue by JoshuaRamirez** [JoshuaRamirez/RoslynMcpServer#847]
## Summary

`get_diagnostics` never exposes Roslyn `Diagnostic.IsSuppressed`, and the product path `Compilation.GetDiagnostics()` **drops** pragma-suppressed compiler diagnostics entirely (they are not returned with `IsSuppressed=true`). Agents therefore cannot tell “this warning was fixed” from “this warning is still present but `#pragma warning disable` / a diagnostic suppressor hid it.”

Verified on master @ `dddf25bee962a80158cb7fda8ceebb5314ca9930` (Release v0.5.2; packages Roslyn **5.0.0**). Label **bug** only (not leftover — open leftover count 0). Leave Dependabot alone.

## Evidence (live Roslyn 5.0.0)

Product contract `DiagnosticInfo` (`GetDiagnosticsResult.cs`) exposes only: `Id`, `Message`, `Severity`, `File`, `Line`, `Column`, `Category`. No `IsSuppressed` (and no include-suppressed switch on `GetDiagnosticsParams`).

Product mapper `GetDiagnosticsOperation` iterates:

```csharp
foreach (var diag in compilation.GetDiagnostics(cancellationToken))
{
    // … severity + file filters …
    var info = new DiagnosticInfo
    {
        Id = diag.Id,
        Message = diag.GetMessage(),
        Severity = diag.Severity.ToString(),
        Category = diag.Descriptor.Category,
        File = file,
        Line = line,
        Column = column
    };
}
```

It never reads `diag.IsSuppressed`. Class comment: “Delegates to Roslyn's Compilation.GetDiagnostics().”

### Live probe — compiler pragma (product API)

| Source | API | CS0219 count | IsSuppressed |
|---|---|---|---|
| plain `int unused = 1;` | `Compilation.GetDiagnostics()` | **1** | `False` |
| `#pragma warning disable CS0219` around same | `Compilation.GetDiagnostics()` | **0** | *(row absent)* |
| same pragma | `SemanticModel.GetDiagnostics()` | **0** | *(row absent)* |
| `SpecificDiagnosticOptions` `CS0219=Suppress` | `Compilation.GetDiagnostics()` | **0** | *(row absent)* |

Public surface: `Diagnostic.IsSuppressed` exists on Roslyn **5.0.0**, but pragma / `ReportDiagnostic.Suppress` filtering means the product’s `GetDiagnostics()` loop never sees those rows at all.

### Live probe — DiagnosticSuppressor (shows `IsSuppressed=true` when requested)

With a small analyzer (`TEST001`) + `DiagnosticSuppressor` and `CompilationWithAnalyzersOptions(reportSuppressedDiagnostics: …)`:

| ReportSuppressedDiagnostics | TEST001 in GetAllDiagnosticsAsync | IsSuppressed |
|---|---|---|
| `true` | present | **`True`** |
| `false` | absent | *(row filtered)* |

Product does not use `CompilationWithAnalyzers` / `reportSuppressedDiagnostics`, and would still drop the flag even if a suppressed diagnostic appeared, because `DiagnosticInfo` has no field for it.

## Why it matters

Agents that call `get_diagnostics` to drive fix loops treat “CS0219 gone” as success. Under a `#pragma warning disable` (or a suppressor), the warning is still in the source model but invisible on the product path — so agents can stop early, miss real debt, or cannot audit intentional suppressions. IDE / Roslyn already know suppression state; the MCP payload does not. Same incomplete-payload failure mode as `#841`–`#846` on this build path, but this ticket is **IsSuppressed / include-suppressed** only.

## Repro sketch

1. `.cs` file with `int unused = 1;` → expect CS0219 via product path.
2. Wrap with `#pragma warning disable CS0219` / `restore`.
3. Call `get_diagnostics` (default severity Warning).
4. Observe: CS0219 row **missing** (not present with `IsSuppressed=true`).
5. Control (Roslyn): same `Compilation.GetDiagnostics()` count=0; `Diagnostic.IsSuppressed` exists but is never populated on a returned pragma-suppressed CS0219 because the API filters first.

## Actual

- `DiagnosticInfo` has no `IsSuppressed`.
- `GetDiagnosticsParams` has no include-suppressed option.
- `GetDiagnosticsOperation` uses only `compilation.GetDiagnostics()`, which omits pragma-suppressed compiler diagnostics instead of returning them marked suppressed.

## Expected

1. Emit `IsSuppressed` (`bool`) from `diag.IsSuppressed` whenever a diagnostic is included in the payload.
2. Decide product behavior for **inclusion**:
   - Prefer an opt-in (e.g. `includeSuppressed`) that surfaces pragma / suppressor-suppressed diagnostics with `IsSuppressed=true` (may require a Roslyn path that reports suppressed diagnostics, such as analyzer pipeline `reportSuppressedDiagnostics`, not only bare `Compilation.GetDiagnostics()`), **or**
   - Document that pragma-suppressed compiler diagnostics are intentionally filtered and only suppressor-marked rows (if ever present) get the flag.
3. Minimum useful fix: payload field + documented inclusion policy; do not pretend absence means “fixed.”

## Reproduction

1. Open a `.cs` file on master @ `dddf25bee962a80158cb7fda8ceebb5314ca9930` with an unused assigned local (CS0219).
2. Call `get_diagnostics` → CS0219 present.
3. Add `#pragma warning disable CS0219` around the local; call again → CS0219 gone from payload (no `IsSuppressed`).
4. Confirm product sources above never map `IsSuppressed`.

## Related / orthogonal

- Named follow-up from `#846` Separate follow-ups item 1 (IsSuppressed).
- Orthogonal to `#841` (IsWarningAsError / WarningLevel / DefaultSeverity).
- Orthogonal to `#842` (HelpLinkUri).
- Orthogonal to `#843` (EndLine / EndColumn).
- Orthogonal to `#844` (Title / Description).
- Orthogonal to `#845` (IsEnabledByDefault / CustomTags).
- Orthogonal to `#846` (AdditionalLocations).
- Orthogonal to `#448` (severityFilter validation).
- No open leftover (label **bug** only; open leftover count 0). Leave Dependabot alone.

## Separate follow-ups (noticed, NOT filed)

1. **`ProgrammaticSuppressionInfo` / suppressor id** — `GetSuppressionInfo(Compilation)` can expose which suppressor/attribute applied; keep off the minimum IsSuppressed boolean unless deliberately batching.
2. **Analyzer diagnostics in `get_diagnostics`** — product today is compilation diagnostics only; loading project analyzers + suppressors is a larger Operate/Design change — do not fold into minimum IsSuppressed emission.
3. Do **not** fold `#841`–`#846` into this minimum IsSuppressed / include-suppressed ticket unless deliberately batching.

## Acceptance criteria

- [ ] `DiagnosticInfo` exposes `IsSuppressed` from `diag.IsSuppressed` for every emitted diagnostic.
- [ ] Documented inclusion policy: either opt-in include of pragma/suppressor-suppressed rows with `IsSuppressed=true`, or explicit docs that pragma-filtered compiler diagnostics remain absent.
- [ ] If include-suppressed is implemented: pragma-disabled CS0219 (or equivalent) appears with `IsSuppressed=true` when the opt-in is on; remains absent (or documented) when off.
- [ ] Regression test(s): get_diagnostics asserts IsSuppressed mapping and, if supported, include-suppressed pragma behavior.
- [ ] Do not treat `#841`–`#846` or suppressor-id metadata as in-scope for the minimum fix unless explicitly batching.

## Suggested fix direction (high-level)

1. Add `IsSuppressed` to `DiagnosticInfo`; set from `diag.IsSuppressed` in `GetDiagnosticsOperation`.
2. Choose inclusion path: keep `GetDiagnostics()` default (filtered) and add opt-in that uses a suppressed-reporting Roslyn entry point, **or** document filtered-by-design for pragma.
3. Add get_diagnostics regression coverage for pragma CS0219 absence vs optional include; leave HelpLinkUri / AdditionalLocations / Title on their own tickets unless deliberately batching.

## Sources

- Product: `src/RoslynMcp.Contracts/Models/GetDiagnosticsResult.cs`, `GetDiagnosticsParams.cs`, `src/RoslynMcp.Core/Query/GetDiagnosticsOperation.cs` @ `dddf25bee962a80158cb7fda8ceebb5314ca9930`
- Packages: `Microsoft.CodeAnalysis.CSharp.Workspaces` / `Workspaces.MSBuild` **5.0.0** (`RoslynMcp.Core.csproj`)
- Named follow-up parent: `#846` Separate follow-ups item 1
- Related but different: `#841`–`#846`, `#448`
- Live probe: Roslyn **5.0.0** — pragma table + DiagnosticSuppressor `ReportSuppressedDiagnostics` table above; product never copies IsSuppressed onto the payload
